Do not define BSDWAIT to avoid error "storage size of ‘w’ isn’t known".

This is an adapted version of the upstream patch taken from here:
https://github.com/tcsh-org/tcsh/commit/4689eb60a74bf13bc146ca3d76e9d7a124ab7b49.patch

From 4689eb60a74bf13bc146ca3d76e9d7a124ab7b49 Mon Sep 17 00:00:00 2001
From: christos <christos>
Date: Fri, 23 Sep 2016 19:17:28 +0000
Subject: [PATCH] Don't define BSDWAIT for linux anymore.

---
 sh.proc.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/sh.proc.c b/sh.proc.c
index 49b199f..874d67c 100644
--- sh.proc.c
+++ sh.proc.c
@@ -47,11 +47,9 @@ RCSID("$tcsh$")
 # define HZ 16
 #endif /* aiws */
 
-#if defined(_BSD) || (defined(IRIS4D) && __STDC__) || defined(__lucid) || defined(__linux__) || defined(__GNU__) || defined(__GLIBC__)
-# if !defined(__ANDROID__)
-#  define BSDWAIT
-# endif
-#endif /* _BSD || (IRIS4D && __STDC__) || __lucid || glibc */
+#if defined(_BSD) || (defined(IRIS4D) && __STDC__) || defined(__lucid)
+# define BSDWAIT
+#endif /* _BSD || (IRIS4D && __STDC__) || __lucid */
 #ifndef WTERMSIG
 # define WTERMSIG(w)	(((union wait *) &(w))->w_termsig)
 # ifndef BSDWAIT
